File: DailyProblem/TwoSumII-InputArrayIsSorted.java
/**
 * LeetCode 167 Two Sum II - Input array is sorted
 */
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        if (numbers == null || numbers.length < 2) {
            return null;
        }
        int left = 0;
        int right = numbers.length - 1;
        while (numbers[left] + numbers[right] != target && left < right) {
            if ((numbers[left] + numbers[right]) < target) {
                left++;
            } else {
                right--;
            }
        }
        return new int[]{left + 1,right + 1};
    }
}

File: TwoSumII-InputArrayIsSorted.go
func twoSum(numbers []int, target int) []int {
	if (numbers == nil || len(numbers) == 0) {
		return nil
	}

	l := 0
	r := len(numbers) - 1
	for (l < r) {
		cur := numbers[l] + numbers[r]
		if (cur < target) {
			l++
		} else if (cur == target) {
			res := []int{l + 1,r + 1}
			return res
		} else {
			r--
		}
	}
	return nil	
}
